homeassistant/components/google/calendar.py

```diff
@@ -105,7 +105,7 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the google calendar platform."""
-    calendar_service = hass.data[DOMAIN][DATA_SERVICE]
+    calendar_service = hass.data[DOMAIN][entry.entry_id][DATA_SERVICE]
```

Comment:
I was doing some manual testing and observed that google calendars between accounts may share the same `calendar_id` -- e.g. every account may have a calendar with an ID like: `addressbook#contacts@group.v.calendar.google.com`

The current unique id logic has a problem as it did not account for this, and it uses the following logic:

f"{calendar_id}-{entity_name}" if num_entities > 1 else calendar_id,

I think what would make more sense is to include the `{config_entry.unique_id}` as a prefix. However, when I tried to change this I now get a bunch of new calendar entities generated.

Is there a prior art example for how we get integrations out of this bad state and clean this up?

Reply:
I went ahead and implemented this as I think i answered my own question.